nautobot/core/filters.py

```diff
@@ -716,8 +715,23 @@ def get_filters(cls):
         """
         filters = super().get_filters()
 
+        if "static_groups" not in filters and getattr(cls._meta.model, "is_static_group_associable_model", False):
```

Comment:
Might make sense to move this out of `BaseFilterSet` and over to a mixin included in `NautobotModelFilterSet` instead, just for self-consistency's sake? Though we have a lot of filtersets that are derived from BaseFilterSet only, especially in `extras`...